packages/shared/src/components/onboarding/FilterOnboardingV4.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { ReactElement, useMemo } from 'react';
+import classNames from 'classnames';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import request from 'graphql-request';
+import useFeedSettings from '../../hooks/useFeedSettings';
+import { RequestKey, generateQueryKey } from '../../lib/query';
+import {
+  GET_ONBOARDING_TAGS_QUERY,
+  GET_RECOMMENDED_TAGS_QUERY,
+  Tag,
+  TagsData,
+} from '../../graphql/feedSettings';
+import { graphqlUrl } from '../../lib/config';
+import { disabledRefetch } from '../../lib/func';
+import { Button } from '../buttons/Button';
+import { AlertColor, AlertDot } from '../AlertDot';
+import { SearchField } from '../fields/SearchField';
+import useDebounce from '../../hooks/useDebounce';
+import { useTagSearch } from '../../hooks';
+import type { FilterOnboardingProps } from './FilterOnboarding';
+import useTagAndSource from '../../hooks/useTagAndSource';
+import { Origin } from '../../lib/analytics';
+
+type OnSelectTagProps = {
+  tag: Tag;
+};
+
+const tagsSelector = (data: TagsData) => data?.tags || [];
+
+export function FilterOnboardingV4({
+  className,
+}: FilterOnboardingProps): ReactElement {
+  const queryClient = useQueryClient();
+
+  const { feedSettings } = useFeedSettings();
+  const selectedTags = useMemo(() => {
+    return new Set(feedSettings?.includeTags || []);
+  }, [feedSettings?.includeTags]);
+  const { onFollowTags, onUnfollowTags } = useTagAndSource({
+    origin: Origin.TagsFilter,
+  });
+
+  const [refetchFeed] = useDebounce(() => {
+    const feedQueryKey = [RequestKey.FeedPreview];
+    queryClient.cancelQueries(feedQueryKey);
+    queryClient.invalidateQueries(feedQueryKey);
+  }, 1000);
+
+  const onboardingTagsQueryKey = generateQueryKey(
```

Comment:
More of a question, how do you feel about this being non objected props?
I personally hate seeing `undefined` all the time, and get worried what will happen if we change the order.

Wondering if we should consider making it a object?

Reply:
My DR for query keys solves it I think.

Reply:
Left some comments, kind of lost track of that, let's maybe propose if for engineering change again :)